parsers.js

```diff
@@ -1,72 +1,99 @@
-            let [key, ...val] = line.split("=");
-            if (Array.isArray(val)){
-                val = val.join("=");
-            }
-            return { Key: key, Value: val };
-        });
-    }
-    if (typeof(value) === "object"){
-        return Object.entries(value).map(([key, val])=>{
-            return { Key: key, Value: val };
-        });
-    }
-    throw "Unsupported tags format!";
+    return value;
+  }
+  if (typeof (value) === "string") {
+    value = value.split("\n").map((line) => line.trim()).filter((line) => line);
+    return value.map((line) => {
+      const [key, ..._val] = line.split("=");
+      let val = _val;
+      if (Array.isArray(val)) {
+        val = val.join("=");
+      }
+      return { Key: key, Value: val };
+    });
+  }
+  if (typeof (value) === "object") {
+    return Object.entries(value).map(([key, val]) => ({ Key: key, Value: val }));
+  }
+  throw new Error("Unsupported tags format!");
 }
 
 module.exports = {
-    boolean : (value) =>{
-        if (value === undefined || value === null || value === '') return undefined;
-        return !!(value && value !=="false")
-    },
-    text : (value) =>{
-        if (value)
-            return value.split('\n');
-        return undefined;
-    },
-    environmentVariables : (value)=>{
-        const parsedEnvironmentVariables = module.exports.text(value);
-        return !parsedEnvironmentVariables ? undefined : parsedEnvironmentVariables.reduce((obj, rawVariable)=>{
-            const [key, value] = rawVariable.split(':');
-            obj[key.trim()] = value.trim();
-            return obj;
-        },{})
-    },
-    number: (value)=>{
-        if (!value) return undefined;
-        const parsed = parseInt(value);
-        if (parsed === NaN) {
-            throw `Value ${value} is not a valid number`
-        }
-        return parsed;
-    },
-    autocomplete: (value)=>{
-        if (!value) return undefined;
-        if (value.id) return value.id;
-        return value;
-    },
-    string: (value)=>{
-        if (!value) return undefined;
-        if (typeof(value) === "string") return value.trim();
-        throw `Value ${value} is not a valid string`;
-    },
-    array: parseArray,
-    tags: parseTags
-}
\ No newline at end of file
+  boolean: (value) => {
+    if (value === undefined || value === null || value === "") {
+      return undefined;
+    }
+    return !!(value && value !== "false");
+  },
+  text: (value) => {
+    if (value) {
+      return value.split("\n");
+    }
+    return undefined;
+  },
+  environmentVariables: (_value) => {
+    const value = { ..._value };
+    const parsedEnvironmentVariables = module.exports.text(value);
+    if (!parsedEnvironmentVariables) {
+      return undefined;
+    }
+
+    return parsedEnvironmentVariables.reduce((_obj, rawVariable) => {
+      const [k, v] = rawVariable.split(":");
+      const obj = { ..._obj };
+      obj[k.trim()] = v.trim();
+      return obj;
```

Comment:
This reassignment here is completely unnecessary. Just do:
```
return {
  ...obj,
  [k.trim()]: v.trim(),
};
```
No need to modify obj in this reduce at all.